lib/ably/auth.rb

```diff
@@ -411,11 +411,15 @@ def auth_header
     end
 
     # Extra headers that may be used during authentication
-    #
+    # spec - RSA7e
     # @return [Hash] headers
-    def extra_auth_headers
-      if client_id && using_basic_auth?
-        { 'X-Ably-ClientId' => Base64.urlsafe_encode64(client_id) }
+    def client_id_header(realtime=false)
+      if options[:client_id] && using_basic_auth?
+        if realtime
+          { 'clientId' => options[:client_id] }
```

Comment:
For realtime connections the `clientId` is passed in the query params, not as a header - see RSA7e1

Reply:
I think by putting the realtime setting in a method that explicitly mentions header (even if its later put in the right place by subsequent code) is going to make things confusing

Reply:
Yeah, it's just a naming convention. I will try to make it more meaningful 👍

Reply:
Updated -> https://github.com/ably/ably-ruby/pull/418/commits/bb4b71f0abe7e17d58d1d10562fb7fd70a66f351

Reply:
This doesn't alleviate my concern - I think we shouldn't be using the word `header` in the method name because sometimes it won't be a header.

Perhaps something like `request_client_id` or similar.

Reply:
wdyt think about `include_cliend_id`

Reply:
maybe `custom_client_id` or `external_client_id`

Reply:
updated -> https://github.com/ably/ably-ruby/pull/418/commits/0f49edecd66af3e27b7fc5efb75c871e0dccf647

Reply:
Fixed https://github.com/ably/ably-ruby/pull/418/commits/1ff7d6b7877362714542a6fecd70654fa6136941

Reply:
Cool, looks good to me — will leave to @AndyTWF to resolve if happy since it's his thread